bindings/java/src/org/sleuthkit/datamodel/ScoringManager.java

```diff
@@ -0,0 +1,259 @@
+	 *
+	 * @param objId Object id.
+	 * @param connection Connection to use for the query.
+	 *
+	 * @return Score, if it is found, Score(UNKNOWN,NONE) otherwise.
+	 *
+	 * @throws TskCoreException
+	 */
+	private Score getAggregateScore(long objId, CaseDbConnection connection) throws TskCoreException {
+		String queryString = "SELECT significance, confidence FROM tsk_aggregate_score WHERE obj_id = " + objId;
+
+		try {
+			db.acquireSingleUserCaseReadLock();
+
+			try (Statement s = connection.createStatement(); ResultSet rs = connection.executeQuery(s, queryString)) {
+				if (rs.next()) {
+					return new Score(Significance.fromID(rs.getInt("significance")), Confidence.fromID(rs.getInt("confidence")));
+				} else {
+					return new Score(Significance.UNKNOWN, Confidence.NONE);
+				}
+			} catch (SQLException ex) {
+				throw new TskCoreException("SQLException thrown while running query: " + queryString, ex);
+			}
+		} finally {
+			db.releaseSingleUserCaseReadLock();
+		}
+	}
+
+	/**
+	 * Inserts or updates the score for the given object.
+	 *
+	 * @param objId Object id of the object.
+	 * @param score  Score to be inserted/updated.
+	 * @param transaction Transaction to use for the update.
+	 *
+	 * @throws TskCoreException
+	 */
+	private void setAggregateScore(long objId, Score score, CaseDbTransaction transaction) throws TskCoreException {
+		CaseDbConnection connection = transaction.getConnection();
+		setAggregateScore(objId, score, connection);
+	}
+
+	/**
+	 * Inserts or updates the score for the given object.
+	 *
+	 * @param objId Object id of the object.
+	 * @param score  Score to be inserted/updated.
+	 * @param connection Connection to use for the update.
+	 *
+	 * @throws TskCoreException
+	 */
+	private void setAggregateScore(long objId, Score score, CaseDbConnection connection) throws TskCoreException {
+
+		String query = String.format(" INTO tsk_aggregate_score (obj_id, significance , confidence) VALUES (%d, %d, %d)",
+				objId, score.getSignificance().getId(), score.getConfidence().getId());
+
+		switch (db.getDatabaseType()) {
+			case POSTGRESQL:
+				query = "INSERT " + query + " ON CONFLICT DO NOTHING"; //NON-NLS
+				break;
+			case SQLITE:
+				query = "INSERT OR IGNORE " + query;
+				break;
+			default:
+				throw new TskCoreException("Unknown DB Type: " + db.getDatabaseType().name());
+		}
+
+		try {
+			db.acquireSingleUserCaseReadLock();
+
+			try (Statement updateStatement = connection.createStatement()) {
+				updateStatement.executeUpdate(query);
+			} catch (SQLException ex) {
+				throw new TskCoreException("Error updating  final score, query: " + query, ex);//NON-NLS
+			}
+
+		} finally {
+			db.releaseSingleUserCaseReadLock();
+		}
+
+	}
+
+	/**
+	 * Recalculate and update the final score of the specified object.
+	 *
+	 * @param objId Object id.
+	 *
+	 * @return Final score of the object.
+	 */
+	public Score recalculateAggregateScore(long objId) throws TskCoreException {
```

Comment:
I'm confused by this method. If we are updating/recalculating the aggregate score of an object every time a new analysis result is created, what is the use case for this method?

Reply:
The updateAggregateScore() is to update the score as each new analysis result is being added - usually during ingest.

The intent for recalculateAggregateScore() is to be used, say if a user marks one of the analysis results  to be invalid/ignored.

Reply:
Wouldn't it be better if invalidating an analysis result caused the AnalysisResult object to call recalculateAggregateScore(), in which case it does not need to be public API? Then the client doesn't have to have special knowledge of how scoring works, can't forget to call recalculateAggregateScore(), and the solution is more OO and less procedural as well?

Also, are we actually implementing this use case at this time, or are we committing to new public API speculatively?